crates/database/src/orders.rs

```diff
@@ -561,6 +572,37 @@ FROM settlements
     sqlx::query_scalar(QUERY).fetch_one(ex).await
 }
 
+pub fn limit_orders_with_outdated_fees(
+    ex: &mut PgConnection,
+    max: DateTime<Utc>,
+) -> BoxStream<'_, Result<FullOrder, sqlx::Error>> {
+    const QUERY: &str = const_format::concatcp!(
+        "SELECT ",
+        ORDERS_SELECT,
+        " FROM ",
+        ORDERS_FROM,
+        " WHERE o.class = 'limit' AND o.surplus_fee_timestamp < $1",
```

Comment:
This should probably ignore cancelled or expired orders to not waste time on them.

Reply:
https://github.com/cowprotocol/services/pull/714/commits/d828e301f071b2e97f6167ee76d630fb244629f7

I assume we don't need to check if the order is invalidated since this only applies to onchain orders, right?

Reply:
If users want to trade native `ETH` on cowswap they had to convert their `ETH` first to `WETH` (wrapping) because we only support ERC20 tokens out of the box. Because this is quite annoying and confusing for many people the RnD team created the `eth flow` where users can interact with a new [smart contract](https://github.com/cowprotocol/ethflowcontract) to let it place orders on their behalf which manages the wrapping for them.
This is fairly new and I wasn't involved that much so I didn't have that on the top of my head but I'm fairly sure we also don't want to compute quotes for those invalidated orders. Also I'm not sure if there is anything that makes limit orders incompatible with the ethflow in some way (then we could even ignore all those orders). I don't think so but let's check with @josojo.

Reply:
In the first iteration, limit orders should only be possible with WETH, not with ETH.

The reason is that for limit orders the probability of a non-settlement is quite high - either price is not reached, or people want to cancel the order and adjust the price. But since order cancellation and price adjustments are quite expensive for ethflow orders - they require an on-chain tx - there would be a significant cost for users to maintain their limit ETH-orders. Hence, we don't wanna include eth-limit orders in the first feature release. That was also discussed with @anxolin

Reply:
https://github.com/cowprotocol/services/pull/714/commits/cee5ba1d9fb7da72269d7f349f3b09426e20dff4